tests/mypetocean_tests.rb

```diff
@@ -0,0 +1,64 @@
+require 'pry'
+class MyPetOceanTest < Minitest::Test
+
+  def setup
+    @school = School.find_or_create_by!(name: 'school')
+    @term = Term.find_or_create_by!(school_id: @school.id, name: 'term', starts_on: '1989-10-01', ends_on: '1999-12-31')
+    @course = Course.find_or_create_by!(name: 'Catbutt & Other Specist Terms', course_code: rand_course_code, term_id: @term.id)
+    @course_student = CourseStudent.find_or_create_by!(course_id: @course.id)
+    @assignment = Assignment.find_or_create_by!(course_id: @course.id, name: 'Destroy C-138', percent_of_grade: '89', due_at: '2017-03-18 21:06:59.001', active_at: '2017-01-19 20:04:59.001')
+    @pre_class_assignment = Assignment.find_or_create_by!(course_id: @course.id, name: 'Destroy C-139', percent_of_grade: '89', due_at: '2017-07-18 21:06:59.001', active_at: '2016-12-18 21:06:59.001')
+    @pre_class_lesson = Lesson.find_or_create_by!(pre_class_assignment_id: @pre_class_assignment.id, name: 'Karl Jaspers')
+    @lesson = Lesson.find_or_create_by!(name: 'Miguel de Unamuno')
+    @reading = Reading.find_or_create_by!(lesson_id: @lesson.id, order_number: '2334', url: 'https://www.youtube.com/watch?v=kytC4OOFeSs')
+    @instructor = User.find_or_create_by!(last_name: 'Archer', first_name: 'Sterling', email: 'phrasing5@dangerzone.com')
+    @course_instructor = CourseInstructor.find_or_create_by(instructor_id: @instructor.id)
+    @assignment = Assignment.find_or_create_by!(course_id: @course.id, name: 'Find the Kitten and Eat It with Copious Cheddar', percent_of_grade: '100', due_at: '2017-12-18 21:06:59.001', active_at: '2017-02-18 21:06:59.001')
+    @assignment_grade = AssignmentGrade.find_or_create_by!(assignment_id: @assignment.id)
+  end
+
+  def test_assignment_ordering
+    correct_order = Assignment.all.sort_by{ |a| [-a.due_at.to_i, -a.active_at.to_i] }
```

Comment:
This can sometimes fail because of the setup above: Essentially, sometimes assignments get created with the same datetimes and sometimes the database breaks those ties in unexpected ways.